src/pages/test-report/cat-be/components/reverse-diagram-modal/reverse-diagram-modal.html

```diff
@@ -0,0 +1,114 @@
+<ion-header>
+  <ion-navbar>
+    <ion-buttons ion-buttons start navPop>
+      <button ion-button center class="nav-close" (click)="closeModal()" id="closeReverseDiagramModal">
+        <ion-icon name="md-close"></ion-icon>Close
+      </button>
+    </ion-buttons>
+    <ion-title>
+      Reversing diagram - articulated vehicle
+    </ion-title>
+  </ion-navbar>
+</ion-header>
+<ion-content>
+  <ion-grid no-padding>
+    <ion-row>
+      <ion-col col-40 class="diagram">
+        <img src="assets/imgs/reverse-diagram.png" alt="Reversing diagram" />
+      </ion-col>
+      <ion-col col-56>
+        <div class="content-container">
+          <h2>Distances</h2>
+          <h3 class="sub-header">All dimensions in metres</h3>
+          <div class="distances">
+            <ion-row>
+              <ion-col>
+                <ion-row>
+                  <ion-col col-50 class="input-label" align-self-center>
+                    <label>Length:</label>
+                  </ion-col>
+                  <ion-col col-42>
+                    <input type="number" step="0.01" id="vehicleLength" class="vehicle-length-input"
+                    [value]="vehicleLength"
+                    (change)="calculateDistanceLength($event.target.value)">
+                  </ion-col>
+                </ion-row>
+              </ion-col>
+              <ion-col>
+                <ion-row>
+                  <ion-col col-44 class="input-label" align-self-center>
+                    <label>Width:</label>
+                  </ion-col>
+                  <ion-col col-42>
+                    <input type="number" step="0.01" id="vehicleWidth" class="vehicle-width-input"
+                    [value]="vehicleWidth"
+                    (change)="calculateDistanceWidth($event.target.value)">
```

Comment:
Feel like this should happen on `(keyup)` rather than `(change)` in the absence of a "Submit" button. Feels confusing to rely on the user focusing away from the field for the calculations to update. Think this requires some thought from service design.

Reply:
Agree It would be better to give the user instant feedback. Will discuss with Nad...